src/passes/flattening/ControlFlowFlattening.cpp

```diff
@@ -74,6 +74,42 @@ void EmitTransition(IRBTy& IRB, AllocaInst* V, BasicBlock* Dispatch,
 
 }
 
+template<class IRBTy>
+void EmitDefaultCaseAssembly(IRBTy& IRB, Triple TT) {
+  auto* FType = FunctionType::get(IRB.getVoidTy(), false);
+  if (TT.isAArch64()) {
+    IRB.CreateCall(FType, InlineAsm::get(
+      FType,
+      R"delim(
+        ldr x1, #-8;
+        blr x1;
+        mov x0, x1;
+        .byte 0xF1, 0xFF;
+        .byte 0xF2, 0xA2;
+      )delim",
+      "",
+      /* hasSideEffects */ true,
+      /* isStackAligned */ true
+    ));
+  } else if (TT.isX86()) {
+    // FIXME: This assembly may not confuse a decompiler
+    IRB.CreateCall(FType, InlineAsm::get(
+      FType,
+      R"delim(
+        nop;
+        .byte 0xF1, 0xFF;
+        .byte 0xF2, 0xA2;
+      )delim",
+      "",
+      /* hasSideEffects */ true,
+      /* isStackAligned */ true
+    ));
+  } else {
+    ExitOnError Exit("Unsupported target for Control-Flow Flattening obfuscation: ");
```

Comment:
Just to let you know, there is a `fatalError` function in `utils.hpp`

Reply:
Thanks for the note. I hadn't seen it, even though it's used a lot throughout the file!